From 09fa0cbdb5c46b8163cf0fe19527dbc1a24fa8c3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 12 Oct 2016 00:44:34 +0100
Subject: [PATCH 7/8] examine: Support multiple directories in the RPATH

As evil as the rpath is, it's infinitely more so when it comes to the
ghc Haskell packages. These use a split list of directories for the rpath,
creating much work for the dynamic linker...

This change ensures we can at least find and support the dependendent libs
throughout these rpaths and not encounter missing symbols at any time.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/examine.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/examine.py b/ypkg2/examine.py
index 872d9e2..ce3408b 100644
--- a/ypkg2/examine.py
+++ b/ypkg2/examine.py
@@ -83,7 +83,7 @@ class FileReport:
             if r:
                 if self.rpaths is None:
                     self.rpaths = set()
-                self.rpaths.add(r.group(1))
+                self.rpaths.update(r.group(1).split(":"))
                 continue
 
             # Match direct needed dependency
-- 
2.10.1

